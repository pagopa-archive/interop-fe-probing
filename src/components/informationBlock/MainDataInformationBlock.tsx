import { Grid } from '@mui/material'
import { InformationContainer } from '@pagopa/interop-fe-commons'
import LaunchIcon from '@mui/icons-material/Launch'
import LockIcon from '@mui/icons-material/Lock'
import { useTranslation } from 'react-i18next'
import { ServiceMainData } from '../../types'
import { ButtonNaked } from '@pagopa/mui-italia'

interface IProps {
  mainData: ServiceMainData
  viewInCatalogue?: () => void
}

const blocksInfo: Array<string> = ['producerName', 'versionNumber', 'eserviceName']

export const MainDataInformationBlock: React.FC<IProps> = ({ mainData, viewInCatalogue }) => {
  const { t } = useTranslation(['detailsPage', 'general'])

  return (
    <>
      <Grid item container direction="column" mb={5}>
        {blocksInfo.map((block) => {
          return block === 'eserviceName' ? (
            <Grid item key={block}>
              <InformationContainer
                key={block}
                label={t(block, { ns: 'detailsPage' }).toUpperCase()}
                content={
                  <ButtonNaked
                    color="primary"
                    size="small"
                    startIcon={<LockIcon />}
                    endIcon={<LaunchIcon />}
                    onClick={viewInCatalogue}
                  >
                    {t('viewInCatalog', { ns: 'detailsPage' })}
                  </ButtonNaked>
                }
              />
            </Grid>
          ) : (
            <Grid item key={block}>
              <InformationContainer
                label={t(block, { ns: 'detailsPage' }).toUpperCase()}
                content={mainData[block as keyof ServiceMainData] as string}
              />
            </Grid>
          )
        })}
      </Grid>
    </>
  )
}
