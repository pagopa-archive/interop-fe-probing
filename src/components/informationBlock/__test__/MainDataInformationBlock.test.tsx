import { describe, expect, test, vi } from 'vitest'
import { render } from '@testing-library/react'
import { MainDataInformationBlock } from '../MainDataInformationBlock'

const mainData = {
  eserviceName: 'Probing test 3',
  producerName: 'Comune di Milano',
  versionNumber: '7',
  pollingFrequency: 5,
}

describe('MainDataInformationBlock', () => {
  test('render component', () => {
    const { container } = render(
      <MainDataInformationBlock mainData={mainData} viewInCatalogue={vi.fn()} />
    )
    expect(container).toMatchSnapshot()
  })
})
