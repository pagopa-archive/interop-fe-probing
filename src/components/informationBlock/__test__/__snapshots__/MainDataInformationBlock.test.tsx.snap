// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainDataInformationBlock > render component 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-15vvrrr-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiStack-root css-19qjufr-MuiStack-root"
      >
        <div
          class="MuiBox-root css-iuqs4u"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            PRODUCERNAME
          </p>
        </div>
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
          >
            Comune di Milano
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiStack-root css-19qjufr-MuiStack-root"
      >
        <div
          class="MuiBox-root css-iuqs4u"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            VERSIONNUMBER
          </p>
        </div>
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
          >
            7
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiStack-root css-19qjufr-MuiStack-root"
      >
        <div
          class="MuiBox-root css-iuqs4u"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            ESERVICENAME
          </p>
        </div>
        <div
          class="MuiBox-root css-1rr4qq7"
        >
          <div
            class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall css-14wetfy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LockIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </span>
              viewInCatalog
              <span
                class="MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LaunchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
